src/jdk.incubator.code/share/classes/jdk/incubator/code/internal/ReflectMethods.java

```diff
@@ -401,6 +402,26 @@ private JCVariableDecl opFieldDecl(Name prefix, long flags, CoreOp.FuncOp op) {
         return opFieldTree;
     }
 
+    private JCMethodDecl opMethodDecl(Name prefix, CoreOp.FuncOp op) {
+
+        var mt = new MethodType(com.sun.tools.javac.util.List.nil(), crSyms.funcOpType, com.sun.tools.javac.util.List.nil(), syms.methodClass);
+        var mn = prefix.append('$', names.fromString("op"));
+        var ms = new MethodSymbol(PUBLIC | STATIC | SYNTHETIC, mn, mt, currentClassSym);
+        currentClassSym.members().enter(ms);
+
+        ListBuffer<JCTree.JCStatement> statements = new ListBuffer<>();
+
+        var opVarInit = make.App(make.Ident(crSyms.opParserFromString), com.sun.tools.javac.util.List.of(make.Literal(op.toText())));
+        var opVar = make.VarDef(new VarSymbol(0, names.fromString("op"), crSyms.opType, ms), opVarInit);
```

Comment:
I believe you can skip the intermediate var? E.g. just cast the result of calling the op parser method?